tests/test_web_sendfile_functional.py

```diff
@@ -240,24 +257,26 @@ async def handler(request):
 
     resp = await client.get("/")
     assert resp.status == 200
-    body = await resp.read()
-    assert b"hello aiohttp\n" == body
-    assert resp.headers["Content-Type"] == "application/pdf"
     assert resp.headers.get("Content-Encoding") == "gzip"
+    assert resp.headers["Content-Type"] == "application/pdf"
+    assert await resp.read() == HELLO_AIOHTTP
     resp.close()
     await resp.release()
     await client.close()
 
 
-async def test_static_file_with_gziped_counter_part_enable_compression(
-    aiohttp_client: Any, sender: Any
+@pytest.mark.parametrize("forced_compression", [None, web.ContentCoding.gzip])
+async def test_static_file_with_encoding_and_enable_compression(
+    hello_txt: pathlib.Path,
+    aiohttp_client: Any,
+    sender: Any,
+    forced_compression: Optional[web.ContentCoding],
```

Comment:
`Optional` is considered confusing, so it'd be nice not to introduce it in favor of being explicit:
```suggestion
    forced_compression: "web.ContentCoding | None",
```

Reply:
We shouldn't use the new syntax, it may cause a SyntaxError if inspected at runtime on Python 3.8/9. I'd just leave these alone and let pyupgrade handle them later. FYI, as far as I'm aware the recommendation was always to use `Optional` until it became deprecated. Now we're just waiting till we can use the new syntax.

Reply:
@Dreamsorcerer that's not entirely accurate. There's a quirk (or two). By default, `web.ContentCoding | None` is problematic, yes. But `"web.ContentCoding | None"` is not.

From the CPython parser's perspective, it's just a string so it does nothing in runtime and you can put anything you want in there, and it won't explode.

When it is being processed by MyPy, though, it allows using newer syntax in the annotations and it works regardless of the target CPython version for as long as MyPy isn't ancient.

Moreover, if we were to put `from __future__ import annotations` at the top, the quotes would be unnecessary and `web.ContentCoding | None` would just work due to the evaluation being postponed as it turns typing definitions in such modules to strings in runtime.

I've been using the future + no quotes for quite a while without any problems. As an example, there's instances of this syntax in yarl and I set up MyPy to test typing against Python 3.7 there — https://results.pre-commit.ci/run/github/64756820/1706701102.QzIu9jkORbOBrIsSqNFdRg.

Reply:
> From the CPython parser's perspective, it's just a string so it does nothing in runtime and you can put anything you want in there, and it won't explode.

When the function is parsed, yes. But, if you inspect the type with get_type() or similar, I think it may evaluate it then, or if using something like Pydantic, then it may try to eval the string if not. It's an edge case, but I don't see any reason to break it. I'm also not sure if pyupgrade would remove the quotes when they are no longer needed, but it'll definitely remove the Optional.

Reply:
I see your point. Though, I think we should rely on what's tested, not on something some people remember to do occasionally. But fine, I won't insist.l

Reply:
I'm in favor of `from __future__ import annotations` as much as possible as the new syntax is so much nicer and more readable. Thats probably for another PR though.